src/Mode.h

```diff
@@ -310,9 +310,24 @@ inline rgba_t reduce_color(const rgba_t color, Mode to_mode) {
       return transparent_color;
     } else {
       rgba_color c(color);
-      c.r >>= 3;
-      c.g >>= 3;
-      c.b >>= 3;
+      if ((c.r & 0x0007) > 0x0003) {
```

Comment:
This can be replaced by `c.r = (c.r + 4) >> 3;`, and so forth for all the other colors.

With such a change, it looks good to me; replacing rounding down with rounding evenly.